packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3177,6 +3177,16 @@ export class ContainerRuntime
 		readAndParseBlob: ReadAndParseBlob,
 		versionId: string | null,
 	): Promise<{ snapshotTree: ISnapshotTree; versionId: string; latestSnapshotRefSeq: number }> {
+		const recoveryMethod = this.mc.config.getString(
+			"Fluid.ContainerRuntime.Test.SummarizationRecoveryMethod",
+		);
+		if (recoveryMethod === "restart") {
+			this._summarizer?.stop("latestSummaryStateStale");
+			const error = new GenericError("Restarting summarizer instead of refreshing");
+			this.closeFn(error);
+			throw error;
```

Comment:
where does this error land? I.e. do we issue a bunch of error logs?
Given this is somewhat normal (or rather - expected) flow, should we have errors in telemetry on this path at all, including container closure error? It's probably fine to have it for now, but we should move toward direction of - error === bug.

Reply:
I wanted a way to stop execution of fetch as that's a network call, so I wrote this throw here. I'm open to any suggestions.

Reply:
Well, I do not want cascade of errors :) I do not know what's the right thing to do here, we might move ahead with that for now, but we will need to solve it IMHO.
The way I'd describe it - ideally, we use exceptions for exceptional cases. Something that is part of normal flow (and I'd argue wrong nack is part of normal flow) should be explicitly handled in code.
We do not follow that line in our repo (driver is a good example where we don't and its really painful), but I think we should reexamine how we deal with code flow here.

Reply:
[AB#4015](https://dev.azure.com/fluidframework/internal/_workitems/edit/3894)